streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -747,6 +748,18 @@ public void setUncaughtExceptionHandler(final Thread.UncaughtExceptionHandler eh
     }
 
     /**
+     * Set the listener which is triggered whenever a {@link StateStore} is being restored in order to resume
+     * processing.
+     *
+     * @param stateRestoreListener The listener triggered when {@link StateStore} is being restored.
+     */
+    public void setStateRestoreListener(final StateRestoreListener stateRestoreListener) {
```

Comment:
One caveat of letting users to set it via code is that, we cannot forbid users to call this function after calling `streams.start()`, in which case the behavior would be bad. Also people can set it multiple times which is also not suggested.

I'm now thinking maybe we should enforce users to set this global listener via configs? cc @dguy @bbejeck @mjsax WDYT.